RyuSocks/Packets/Auth/UsernameAndPassword/UsernameAndPasswordRequest.cs

```diff
@@ -0,0 +1,23 @@
+using System.Text;
+
+namespace RyuSocks.Packets.Auth.UsernameAndPassword
+{
+    public class UsernameAndPasswordRequest : IPacket
+    {
+        public byte Version { get; set; }
+        public byte UsernameLength;
+        public string Username;
+        public byte PasswordLength;
+        public string Password;
+        public void FromArray(byte[] array)
+        {
+            Version = array[0];
+            UsernameLength = array[1];
+            Username = Encoding.ASCII.GetString(array[2..(UsernameLength + 1)]);
+            PasswordLength = array[(UsernameLength + 2)];
+            Password = Encoding.ASCII.GetString(array[(UsernameLength + 2)..(PasswordLength + 3)]);
```

Comment:
```suggestion
            Username = Encoding.ASCII.GetString(array[2..(2 + UsernameLength)]);
            PasswordLength = array[2 + UsernameLength];
            Password = Encoding.ASCII.GetString(array[(2 + UsernameLength + 1)..((2 + UsernameLength + 1) + PasswordLength)]);
```
Feel free to modify this, just wanted to have the reference here.